cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -281,35 +299,24 @@ public boolean accept(String className, URL classUrl, URL classPathUrl) {
       }
     };
 
-    Set<File> hBaseTableDeps = traceDependencies(HBaseTableUtilFactory.getHBaseTableUtilClass().getName(),
-                                                 usingCL, classAcceptor);
+    Set<File> hBaseTableDeps = traceDependencies(usingCL, classAcceptor, tmpDir,
+                                                 HBaseTableUtilFactory.getHBaseTableUtilClass().getName());
 
     // Note the order of dependency jars is important so that HBase jars come first in the classpath order
     // LinkedHashSet maintains insertion order while removing duplicate entries.
     Set<File> orderedDependencies = new LinkedHashSet<>();
     orderedDependencies.addAll(hBaseTableDeps);
-    orderedDependencies.addAll(traceDependencies(DatasetService.class.getName(),
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies("co.cask.cdap.hive.datasets.DatasetStorageHandler",
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies("co.cask.cdap.hive.datasets.StreamStorageHandler",
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies("org.apache.hadoop.hive.ql.exec.mr.ExecDriver",
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies("org.apache.hive.service.cli.CLIService",
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies("org.apache.hadoop.mapred.YarnClientProtocolProvider",
-                                                 usingCL, classAcceptor));
-    orderedDependencies.addAll(traceDependencies(RecordFormats.class.getName(),
-                                                 usingCL, classAcceptor));
-
-    // Needed for - at least - CDH 4.4 integration
-    orderedDependencies.addAll(traceDependencies("org.apache.hive.builtins.BuiltinUtils",
-                                                 usingCL, classAcceptor));
-
-    // Needed for - at least - CDH 5 integration
-    orderedDependencies.addAll(traceDependencies("org.apache.hadoop.hive.shims.Hadoop23Shims",
-                                                 usingCL, classAcceptor));
+    orderedDependencies.addAll(traceDependencies(usingCL, classAcceptor, tmpDir,
+                                                 DatasetService.class.getName(),
+                                                 "co.cask.cdap.hive.datasets.DatasetStorageHandler",
```

Comment:
Even it is a cdap class? Where does it come from?

Sent from my iPhone

> On Aug 28, 2015, at 11:17 AM, Alvin Wang <notifications@github.com> wrote:
> 
> In cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java:
> 
> > -                                                 usingCL, classAcceptor));
> > -    orderedDependencies.addAll(traceDependencies("org.apache.hadoop.mapred.YarnClientProtocolProvider",
> > -                                                 usingCL, classAcceptor));
> > -    orderedDependencies.addAll(traceDependencies(RecordFormats.class.getName(),
> > -                                                 usingCL, classAcceptor));
> > -
> > -    // Needed for - at least - CDH 4.4 integration
> > -    orderedDependencies.addAll(traceDependencies("org.apache.hive.builtins.BuiltinUtils",
> > -                                                 usingCL, classAcceptor));
> > -
> > -    // Needed for - at least - CDH 5 integration
> > -    orderedDependencies.addAll(traceDependencies("org.apache.hadoop.hive.shims.Hadoop23Shims",
> > -                                                 usingCL, classAcceptor));
> > +    orderedDependencies.addAll(traceDependencies(usingCL, classAcceptor, tmpDir,
> > +                                                 DatasetService.class.getName(),
> > +                                                 "co.cask.cdap.hive.datasets.DatasetStorageHandler",
> Had to use plain strings since DefaultStorageHandler isn't present in the class path here.
> 
> —
> Reply to this email directly or view it on GitHub.
>